packages/ciphernode/evm/src/registry_filter_sol.rs

```diff
@@ -0,0 +1,134 @@
+use crate::helpers::{create_provider_with_signer, SignerProvider};
+use actix::prelude::*;
+use alloy::{
+    primitives::{Address, Bytes, U256},
+    rpc::types::TransactionReceipt,
+    signers::local::PrivateKeySigner,
+    sol,
+};
+use anyhow::Result;
+use enclave_core::{
+    BusError, E3id, EnclaveErrorType, EnclaveEvent, EventBus, OrderedSet, PublicKeyAggregated,
+    Subscribe,
+};
+use std::sync::Arc;
+
+sol!(
+    #[sol(rpc)]
+    NaiveRegistryFilter,
+    "../../evm/artifacts/contracts/registry/NaiveRegistryFilter.sol/NaiveRegistryFilter.json"
```

Comment:
@ryardley, I don't think this needs to block merging, but I am keen to understand what is missing from `IRegistryFilter.sol` that required you to use`NaiveRegistryFilter.sol` here? I assume it's the `publishCommittee()` method. In which case, I think we should perhaps include this in `IRegistryFilter.sol`, may also imply restructuring it to make it more generic.

Perhaps something like this, essentially an extra `data` field that can be used to pass in whatever other additional data (proofs, for example) that a given registry may need.

```solidity
    function publishCommittee(
        uint256 e3Id,
        address[] memory nodes,
        bytes memory publicKey,
        bytes memory data
    ) external;
```